### PR TITLE
Documented dataset format

### PR DESCRIPTION
**Pull request by smahes20**
Under the docs folder, created a file to document the dataset format and the datatypes of each field.

### CODE REVIEW

File: stock_scrapper.py [hunk 1]
```diff
@@ -1,19 +1,60 @@
 """
     File name: stock_scrapper.py
-    Author: Jonathan Birkey
-    Email: jbirke2@illinois.edu
-    Date created: 29 Oct 2021
-    Python Version: 3.10.0
-    Notes: S&P 500 company list pulled from https://datahub.io/core/s-and-p-500-companies#data
+        Author: Jonathan Birkey
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:3]: can you single tab this section?

File: stock_scrapper.py [hunk 2]
```diff
@@ -1,19 +1,60 @@
...
+        Task: Curate the collected data and store it in a JSON file. 
+
 """
 import csv
 import datetime
 import numpy as np
 import pandas as pd
 import time
+import json
+import os
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:23]: please make sure imports or in alphabetical order

File: stock_scrapper.py [hunk 3]
```diff
@@ -1,19 +1,60 @@
...
+        for item in df.values:
+            emptyjs.update(json.loads(template%tuple(item.tolist())))
+        return emptyjs
+    except BaseException as err:
+        print(f'Something bad happened!: {err=}')
+        return
 
-def get_sp500_tickers(file: str) -> list[str]:
+
+def get_fomo_tickers(file: str) -> list:
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:57]: `def get_fomo_tickers(file: str) -> list[str]:`

File: stock_scrapper.py [hunk 4]
```diff
@@ -23,30 +64,44 @@ def get_sp500_tickers(file: str) -> list[str]:
...
 
 
 def main():
-    file_path = 'data/sp_500_constituents_as_of_20211022.csv'
-    tickers = get_sp500_tickers(file_path)
-    ticker = 'MMM'
+    file_path = 'data/fomo_etf_as_of_20211031.csv'
+
+    #if the dataset file exists then delete file to avoid overwritting of data 
+    output_path = 'data/Team_yoda_dataSet.json'
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:81]: please don't mix snake case and camel case. I would recommend just over writing the .json file I created so we don't have two json files

File: stock_scrapper.py [hunk 5]
```diff
@@ -23,30 +64,44 @@ def get_sp500_tickers(file: str) -> list[str]:
...
+    if os.path.exists(output_path):
+        os.remove(output_path)
+
+    tickers = get_fomo_tickers(file_path)
     period1 = int(time.mktime(datetime.datetime(2020, 1, 1, 23, 59).timetuple()))
     period2 = int(time.mktime(datetime.datetime(2020, 12, 31, 23, 59).timetuple()))
     interval = '1d'  # daily = 1d, weekly = 1wk, monthly = 1m
+    file_data = {} #initial empty dict to populate
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:89]: PEP8 should be two spaces after code before comment and then one space after the #

File: stock_scrapper.py [hunk 6]
```diff
@@ -23,30 +64,44 @@ def get_sp500_tickers(file: str) -> list[str]:
...
+    for ticker in tickers:
+        query_str = f'https://query1.finance.yahoo.com/v7/finance/download/{ticker}?period1={period1}&' \
+                    f'period2={period2}&interval={interval}&events=history&includeAdjustedClose=true'
+        df = pd.read_csv(query_str)
+        calculate_daily_change_rate(df)
+        jsonval = build_stock_json(df)
+        file_data[ticker] = jsonval
 
-    # TODO: calculate change and change rate for every ticker in the tickers list and save it as a single dataset
+    # print(json.dumps(file_data,indent=4))    ## to view it in json format with indentation
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:99]: please remove dead code

File: stock_scrapper.py [hunk 7]
```diff
@@ -23,30 +64,44 @@ def get_sp500_tickers(file: str) -> list[str]:
...
-                f'period2={period2}&interval={interval}&events=history&includeAdjustedClose=true'
-
-    df = pd.read_csv(query_str)
-    calculate_daily_change_rate(df)
+    for ticker in tickers:
+        query_str = f'https://query1.finance.yahoo.com/v7/finance/download/{ticker}?period1={period1}&' \
+                    f'period2={period2}&interval={interval}&events=history&includeAdjustedClose=true'
+        df = pd.read_csv(query_str)
+        calculate_daily_change_rate(df)
+        jsonval = build_stock_json(df)
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:96]: PEP8 please use snake case `json_val`

File: stock_scrapper.py (hunk 3 above)
Comment (smahes20) [reply stock_scrapper.py:57]: I had changed this on my end as it was giving me an error and forgot to change it back.

File: stock_scrapper.py (hunk 2 above)
Comment (smahes20) [reply stock_scrapper.py:23]: will do.

File: stock_scrapper.py (hunk 6 above)
Comment (smahes20) [reply stock_scrapper.py:99]: will do

File: stock_scrapper.py [hunk 8]
```diff
@@ -5,15 +5,56 @@
...
 import numpy as np
+import os
 import pandas as pd
 import time
 
 
-def get_sp500_tickers(file: str) -> list[str]:
+def build_stock_json(df: pd.DataFrame):
+    try:
+        emptyjs = {}
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:28]: PEP8: please use snake case `empty_js`

File: stock_scrapper.py [hunk 9]
```diff
@@ -5,15 +5,56 @@
...
 import datetime
+import json
 import numpy as np
+import os
 import pandas as pd
 import time
 
 
-def get_sp500_tickers(file: str) -> list[str]:
+def build_stock_json(df: pd.DataFrame):
```
Comment (Jonathan-Birkey) [inline stock_scrapper.py:26]: Please add return type
`def build_stock_json(df: pd.DataFrame) -> dict:`

**Review by Jonathan-Birkey — APPROVED** [review]

**Comment by Jonathan-Birkey** [comment]
Looks good to me! merge when ready!

**Pull request merged**